include/string_span.h

```diff
@@ -109,26 +109,26 @@ inline basic_string_span<T, dynamic_range> ensure_z(T* const & sz, std::ptrdiff_
 // TODO (neilmac) there is probably a better template-magic way to get the const and non-const overloads to share an implementation
 inline basic_string_span<char, dynamic_range> ensure_z(char* const& sz, std::ptrdiff_t max)
 {
-    auto len = strnlen(sz, max);
+    auto len = strnlen(sz, static_cast<size_t>(max));
     Ensures(sz[len] == 0);
```

Comment:
That's a good point.  However, I'm not sure how I can easily use `narrow_cast`; gsl.h defines `narrow_cast`, but I think there's a circular header dependency so if I change to `narrow_cast`, it's trying to use it before it's declared.

Perhaps if `narrow_cast` lived in it's own header...?